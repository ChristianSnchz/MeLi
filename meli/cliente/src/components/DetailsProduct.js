
/**
 * Componente Detalle de producto 
 * Encargado de consultar por el id del producto mediante la funcion getProduct
 * 
 * @function getProduct
 * @param {string} id
 *  Toma un ID y consulta al servidor por el item devuelve un arreglo con el item y la descripcion
 * 
*/

import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import '../styles/styles.scss';
import axios from 'axios';

export const DetailsProduct = () => {

  const history = useHistory();

  const [product, setproduct] = useState();

  useEffect(() => {

    const { location } = history;
    const { pathname } = location;
    const id = pathname.substring(7, pathname.length );    
    getProduct(id);
        
  }, [history]);


  const getProduct = async (id) => {
    const apiurl = 'http://localhost:4000/api';    
    const uri = `${apiurl}/items/${id}`;    
    const response = await axios.get(uri);
    const { data } = response;  
    setproduct(data.item);
  }


  return (
    <>
    { product&& <div  className="details_container_two">
        <div className="details_container" >

          <div style={{ display:'flex'  }} >

            <img src={product.picture}
              className="detail_image"
              alt="Portada"
            />
            
            <div className="detail_box" >
              <p className="detail_subtittle" >{product.condition}</p>
              <p className="detail_tittle" > {product.title} </p>
              <p className="detail_price" > {product.price.currency} {product.price.amount} <span style={{ fontSize:'20px' }} >00</span>   </p>
              
              <button className="detail_button">
                Comprar
              </button>
            </div>
          </div>

          <div >
            <p className="detail_description_tittle"> Descripci√≥n del producto </p>

            <p className="detail_description" >
              {product.description}
            </p>

          </div>
        </div>
      </div>}          
    </>
  )
}
