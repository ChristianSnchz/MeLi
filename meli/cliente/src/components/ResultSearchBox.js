
/**
 * Componente Resultados de Busqueda
 * 
 * Encargado de mostrar dinamicamente los resultados obtenidos por la consulta al servidor
 * de la query obtenida por el Header (SearchBox) Mediante la funcion SearchService 
 * 
 * 
 * @function SearchService 
 * @param {string} param query de la caja de busqueda del header
 * 
 */

import React, { useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import '../styles/styles.scss';
import shipping from '../assets/ic_shipping.png';
import axios from 'axios';

export const ResultSearchBox = () => {
    
  const [itemsSearch, setitemsSearch] = useState([]);
  const [categories, setCategories] = useState([]);
  const history = useHistory();

  useEffect(() => { 
    const { location } = history;
    const { search } = location;
    if(search === '?search='){
      history.push('/');
    }    
    
    searchService(search.substring( 8,search.length ));            
  }, [history.location.search]);


  const searchService = async ( param ) => {    
    const apiurl = 'http://localhost:4000/api';    
    const uri = `${apiurl}/items?q=${param}`;    
    const response = await axios.get(uri);
    const { data } = response;  
    console.log(data);
    setCategories(data.categories);
    setitemsSearch(data.items);      
  }

  const goToDetail = (item) => {
    console.log(item.id);
    history.push(`/items/${item.id}`); 
  
  }

  return (
    <>    
    <div style={{ marginTop:'1rem',  display:'flex', marginLeft:'16rem' }}>      
          {categories.map( (category) => {            
            if(category === categories[categories.length -1] ){
              return <p style={{marginLeft:'1rem'}} > { `${category}`}</p>
            }else{
              return <p style={{marginLeft:'1rem'}}> { `${category}   >`}</p>
            }            
          }) 
          }      
    </div>
      {       
        itemsSearch.map( (item) => {
            return  <div key={item.id} className="items_container_two" > 
              <div className="items_container" onClick={ () => goToDetail(item) } >                
                    <img 
                      src={item.picture}
                      className="items_img"
                      />

                  <div>
                    <div style={{ display:'flex' }}  >
                      <h3 > {item.price.currency} { item.price.amount } </h3> 
                      { item.free_shipping && <img src={shipping}  style={{height:'20px', marginLeft:'0.8rem', marginTop:'1.1rem' }} /> }                      
                    </div>
                    <p> {item.title } </p>                    
                  </div>

                  <div className="items_location" >
                    <p> Ubicacion </p>
                  </div>

              </div>
            </div>
        })       
      }      
    </>
  )
}
