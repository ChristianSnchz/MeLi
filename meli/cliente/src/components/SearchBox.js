import React, { useState } from 'react';
import "../styles/styles.scss";
import logo from '../assets/Logo_ML.png';
import lupa from '../assets/ic_Search.png';
import { useHistory } from 'react-router-dom';


export const SearchBox = () => {

  const history = useHistory();

  const [search, setsearch] = useState('');

  const searching = (e) => {
    e.preventDefault();
    history.push(`/items?search=${search}`);
    
  }

  const handleInputChange = (e) => {        
    setsearch(e.target.value);
  }

  const homeBack = ()=> { history.push('/'); }


  return (
    <>

    <div className="search_box">      
          <form onSubmit={searching} className="container_search" >

            <div onClick={ homeBack }>
              <img  src={logo} alt="logo" className="logo_search" />
            </div>

            <input type="text" 
                  value={search} 
                  onChange={ handleInputChange  }
                  name="search" 
                  className="input_search" 
                  placeholder="Nunca dejes de buscar" />     

            <button  className="button_search" >
              <img  src={lupa} alt="lupa" />
            </button>

          </form>      
    </div>
      
    </>
  )
}
