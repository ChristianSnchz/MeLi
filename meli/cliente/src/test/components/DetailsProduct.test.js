import React from 'react';
import { mount } from "enzyme";       
import { MemoryRouter } from 'react-router-dom';
import { DetailsProduct } from '../../components/DetailsProduct';


describe('Test in ResultSearchBox component', () => {
  
    test('should match with snaphot', () => {

    const wrapper = mount(      
        <MemoryRouter >
          <DetailsProduct />
        </MemoryRouter>                   
    );
    expect( wrapper ).toMatchSnapshot();    
  });
    
})
