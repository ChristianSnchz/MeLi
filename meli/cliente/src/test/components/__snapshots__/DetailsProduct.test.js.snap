// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in ResultSearchBox component should match with snaphot 1`] = `""`;
